knimin/lib/parse.py

```diff
@@ -94,3 +95,84 @@ def parse_qpcr_object(contents):
         result[r, c] = value
 
     return result
+
+
+def _find_section(lines, label):
+    """Find a section, eg [EXCEPTIONS]
+
+    Parameters
+    ----------
+    lines : list of str
+        The data to search over
+    label : str
+        The section header to search for
+
+    Returns
+    -------
+    int
+        The index of the header
+    int
+        The index of the first empty line following the section, or the index
+        position of the end of the file.
+
+    Raises
+    ------
+    ValueError
+        If the section is not found
+    """
+    start = -1
+    end = len(lines)
+    for idx, line in enumerate(lines):
+        if line.startswith(label):
+            start = idx
+            break
+
+    if start == -1:
+        raise ValueError('%s section appears to be missing' % label)
+
+    for idx, line in enumerate(lines[start:], start + 1):
+        if not line or line == '':
+            end = idx
+            break
+
+    return start, end
+
+
+def parse_echo(data):
+    """Parse the Echo output file
+
+    Parameters
+    ----------
+    data : str
+        A str representation of the file
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame composed of the [EXCEPTIONS] section
+    pd.DataFrame
+        A DataFrame composed of the [DETAILS] section
+
+    Raises
+    ------
+    ValueError
+        If the header is missing
+        If the exceptions section is missing
+        If the details section is missing
+    """
+    data = data.splitlines()
```

Comment:
If `data` has extra trailing empty new lines, this conversion will leave some empty elements in the end. Maybe  do `data.rstrip('\n')` beforehand?